src/legacy/ui/public/field_editor/field_editor.js

```diff
@@ -331,6 +333,57 @@ export class FieldEditorComponent extends PureComponent {
     );
   }
 
+  renderTypeConflict() {
+    const { field } = this.state;
+    if(!field.conflictDescriptions) {
+      return null;
+    }
+    const labelText = (
+      <EuiCallOut
+        color="warning"
+        title={<FormattedMessage id="common.ui.fieldEditor.multiTypeLabel" defaultMessage="Multiple type field"/>}
+        size="s"
+        iconType="alert"
+      >
+        <FormattedMessage
+          id="common.ui.fieldEditor.multiTypeLabelDesc"
+          defaultMessage="The type of this field changes across indices. It is unavailable for many analysis functions."
+        />
+      </EuiCallOut>
+    );
+
+    const columns = [{
+      field: 'type',
+      name: 'Type',
+      width: '100px',
+    }, {
+      field: 'indices',
+      name: 'Index Names'
+    }];
+
+    const items = Object.keys(field.conflictDescriptions).map(type => {
```

Comment:
:art: Minor code style nitpick: Since we anyway need both the key as the value, we could `map` over `entries` instead of `keys` (also we could easily get rid of the `return` here).

```js
const items = Object.entries(field.conflictDescriptions).map(([type, indices]) => ({
  type,
  indices: indices ? indices.join(', ') : [],
}));
```

Reply:
Among several changes, i've drawn 🖌️ it this way now:
```
const items = Object
      .entries(conflictDescriptions)
      .map(([type, indices]) => ({
        type,
        indices: Array.isArray(indices) ? indices.join(', ') : ''
      }));
```